syft/execution/type_wrapper.py

```diff
@@ -0,0 +1,261 @@
+import syft as sy
+from syft.workers.abstract import AbstractWorker
+from warnings import warn
+from typing import Union
+from syft_proto.execution.v1.type_wrapper_pb2 import NestedTypeWrapper as NestedTypeWrapperPB
+from syft_proto.execution.v1.type_wrapper_pb2 import ClassType as ClassTypePB
+
+class NestedTypeWrapper:
+    """
+        Class for input type serialization and type checking for nested structures.
+    """
+    def __init__(self, nested_type=None):
```

Comment:
I think the `if` would be better in the `serialize_nested_type` method
Also, I'm not sure the pattern of instantiating a class and doing what you want to do at the same time is the most intuitive, I'll try to think about it but I guess it's ok